components/brave_welcome_ui/main_container.tsx

```diff
@@ -47,9 +48,22 @@ function MainContainer () {
     mainEl = <HelpImprove />
   }
 
+  if (viewType === ViewType.DefaultBrowser) {
+    mainEl = <Welcome />
+  }
+
+  const onBackgroundImgLoad = () => {
+    setViewType(ViewType.DefaultBrowser)
+  }
+
   return (
-    <Background static={!shouldPlayAnimations}>
-      {mainEl}
+    <Background
+      static={!shouldPlayAnimations}
+      onLoad={onBackgroundImgLoad}
+    >
+      <React.Suspense fallback={<div>Loading...</div>}>
```

Comment:
Perhaps just the simple loading spinner that we have. It comes from brave-ui. I wouldn't have this committed since 1) it might actually show (and it isn't styled) and 2) it's not translated.

Here's an example of the loading spinner being used https://github.com/brave/brave-core/blob/63be302a0931552f3651748ad7fea190dd5b8f18/components/brave_new_tab_ui/components/loading.tsx

Reply:
added

Reply:
This results in a flash of the loader which disrupts the user experience.

Reply:
To solve the issue where the loader briefly flashes on the screen before the component finishes loading, we can add a delay of 100ms before rendering the loader.

Reply:
how did you resolve this?